app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataValuePresenter.kt

```diff
@@ -282,24 +176,40 @@ class DataValuePresenter(
     }
 
     fun onSaveValueChange(cell: TableCell) {
+        runBlocking {
+            saveValue(cell)
+            view.onValueProcessed()
```

Comment:
I think updateData/saveData ended being called on the main thread accessing the repository??

Reply:
Nope. saveValue is a suspend function that runs on io. I need it runBlocking so that onValueProcessed get called after the saveValue. I've made some tests and the UI is not blocked at all.

Reply:
oh, did not see the withContext(dispatcherProvider.io()) { which was in the same line

my fault